### ISSUE TITLE
FreeOTP working but backup/restore missing password

### ISSUE THREAD

**Issue by mihaime** [freeotp/freeotp-android#373]
I have FreeOTP running and all is good but if I try to do a backup and restore on another phone, I do not have the password.
Is there any way to make this work out?
I am having access to the app and QR codes are working.
Or is this an unsupported scenario?

**Comment by Truyn** [comment]
Likewise, complete crap, I can’t transfer the backup, I don’t remember the password and I don’t even remember entering it. Why are there no other methods? Why this insanity, someone asked me if I needed an additional password

**Comment by mariush444** [comment]
the same to me.
I make backup - the apk doesn't ask to enter a password but during restore there is a message to enter password.
But which one ?

**Comment by VincentSC** [comment]
Duplicate of #367 #321 #348 #344 #283 and probably several more.

The app's "solution" is just to blame the user for not remembering ever entering a password years ago.

Assume this will not be fixed any time soon, and better put the time into replacing every OTP-code you use...

**Comment by mihaime** [comment]
I already implemented the final fix for this.
Remove the app and replace it with something else :)
One time recovering all accounts has been quite a life lesson. Never again.

**Comment by mariush444** [comment]
> I already implemented the final fix for this. Remove the app and replace it with something else :) One time recovering all accounts has been quite a life lesson. Never again.

Yes, it is very very good advice. It is not ironic, it is security reason. If developer can't understand if-then-else  that means it is time to escape.
Over

**Comment by Stan-Green-PM** [comment]
Yes, I did know my backup password and I was able to use it to restore to my new phone. So lesson learned is to know your passwords!

**Comment by VincentSC** [comment]
Maybe you can import the file into Aegis, to see if the export or the import has the bugs?

**Comment by Stan-Green-PM** [comment]
Aegis cannot import the vault from FreeOTP:
com.beemdevelopment.aegis.importers.DatabaseImporterException: com.beemdevelopment.aegis.vault.VaultFileException: org.json.JSONException: Value

**Comment by VincentSC** [comment]
Then export is broken

**Comment by Stan-Green-PM** [comment]
UPDATE: I found out that the one token that was not restored cannot be copied to a new device. (I don't know the tech details here.) I had to re-enroll the token. All the other tokens are working. So, I don't think any "bugs" exist in the backup/restore process.

**Comment by jessicachitas** [comment]
Has this issue been fixed? I got a new phone and it transferred all of my otps but they will not activate anymore. They will on my old phone still but not the new one.

**Comment by mihaime** [comment]
In my case it did as I changed the app to one that has better support and developers that maintain it and reply to issues.
Having had to reset all my MFAs once has been the limit of what I could bear.

**Comment by mariush444** [comment]
confirm. I've  choosen different app also. Someone here adviced Aegis app. 
It is good for me too.

**Comment by zrahash** [comment]
on same boat. first I thought i forgot the password, so i tried all passworld , did not work. I tried uninstall and reinstall in a new phone, export backup and tried to import. strangely it ask for password. there was no password set on entire instalaltion/export process. so its a bug, and after gooogling looks like everyone is having same issue.  crap.

**Comment by hydratester** [comment]
I was on the same boat. I have a json backup created by FreeOTP about a month ago. FreeOTP failed to import it on a new phone due to the pw. I tried Aegis and it succeeded. I used the "FreeOTP+ (JSON)" option. No pw was requested and the codes work now. Thanks @VincentSC for mentioning it.
Hope this helps someone else

**Comment by TheRaven500** [comment]
Same here. Restore does not work. The solution was simple -> Switch the App 🙈
